src/Tasks/Common/FileUtilities.MetadataReader.cs

```diff
@@ -21,31 +22,60 @@ namespace Microsoft.NET.Build.Tasks
 {
     static partial class FileUtilities
     {
+        private static Dictionary<string, (DateTime LastKnownWriteTimeUtc, Version Version)> s_versionCache = new(StringComparer.OrdinalIgnoreCase /* Not strictly correct on *nix. Fix? */);
```

Comment:
Do we need to worry about being thread-safe here?  I'm guessing not because MSBuild doesn't run tasks in parallel in the same process.

Reply:
Correct; MSBuild should never run the same task concurrently in a process.